examples/medplum-chart-demo/src/bots/core/general-encounter-note.ts

```diff
@@ -0,0 +1,211 @@
+} from '@medplum/fhirtypes';
+import {
+  calculateBMI,
+  ClinicalImpressionData,
+  ConditionData,
+  createClinicalImpressionEntry,
+  createConditionEntries,
+  createObservationEntries,
+  GeneralObservationData,
+  handleBloodPressure,
+} from './bot-utils';
+
+export async function handler(event: BotEvent<QuestionnaireResponse>, medplum: MedplumClient): Promise<Bundle> {
+  // Parse the answers from the QuestionnaireResponse
+  const response = event.input;
+  const answers = getQuestionnaireAnswers(response);
+  // Get the linked encounter and the user who initiated the bot
+  const encounter = await medplum.readReference(response.encounter as Reference<Encounter>);
+  const user = medplum.getProfile() as Practitioner;
+
+  // Reason for visit is a required answer, so if it is missing throw an error
+  if (!answers['reason-for-visit']?.valueCoding) {
+    throw new Error('Must provide a reason for the visit');
+  }
+
+  // Parse the answers into more easily usable objects
+  const observationData: GeneralObservationData = {
+    bloodPressure: {
+      diastolic: answers['diastolic']?.valueInteger,
+      systolic: answers['systolic']?.valueInteger,
+    },
+    height: answers['height']?.valueQuantity,
+    weight: answers['weight']?.valueQuantity,
+    hotFlash: answers['hot-flashes']?.valueBoolean,
+    moodSwings: answers['mood-swings']?.valueBoolean,
+    vaginalDryness: answers['vaginal-dryness']?.valueBoolean,
+    sleepDisturbance: answers['sleep-disturbance']?.valueBoolean,
+    selfReportedHistory: answers['self-reported-history']?.valueString,
+    date: answers['date']?.valueDateTime,
+  };
+
+  if (observationData.height && observationData.weight) {
+    observationData.bmi = calculateBMI(observationData.height, observationData.weight);
+  }
+
+  const conditionData: ConditionData = {
+    reasonForVisit: answers['reason-for-visit']?.valueCoding,
+    problemList: answers['problem-list']?.valueBoolean || false,
+  };
+
+  const clinicalImpressionData: ClinicalImpressionData = {
+    assessment: answers['assessment']?.valueString,
+  };
+
+  // Take the objects and create bundle entries from the data for each resource
+  const observationEntries = createObservationEntries(observationData, encounter, user, createObservationEntry);
+  const conditionEntries = createConditionEntries(conditionData, encounter, user);
+  const clinicalImpressionEntries = createClinicalImpressionEntry(clinicalImpressionData, encounter, user);
+
+  // Create an array of all entries
+  const entry = [...observationEntries, ...conditionEntries];
+  if (clinicalImpressionEntries) {
+    entry.push(clinicalImpressionEntries);
+  }
+
+  // Create the bundle
+  const encounterNoteBatch: Bundle = {
+    resourceType: 'Bundle',
+    type: 'batch',
+    entry,
+  };
+
+  // Execute the batch to create all resources at once
+  const responseBundle = await medplum.executeBatch(encounterNoteBatch);
+  return responseBundle;
+}
+
+function createObservationEntry(
```

Comment:
Rather than having this callback, a better model might be:
- In each bot, have a function that generates a list of _partial_ observations based on the QuestionnaireResponse.
- Then we pass this array into the `createObservationEntries` function to wrap them